### PR TITLE
make the way version.h is included standard

### PR DESCRIPTION
**Pull request by kinkie**
include guards for version.h are in squid.h
Change this to use the standard approach

### CODE REVIEW

**Comment by rousskov** [comment]
I am abstaining. I do not know what `sub-packages` are. Perhaps this is related to how Squid is packaged? I am asking @yadij to double check this.

**Comment by kinkie** [comment]
That line was last changed 12 years ago, and the HAVE_SQUID symbol is defined nowhere. It's dead code

**Comment by rousskov** [comment]
I was worried that something that packages Squid defines `HAVE_SQUID` to provide custom versions (despite `snmp.h` including `versions.h` unconditionally). That is why I asked Amos to double check. I did not mean to imply that there is something wrong.

**Comment by kinkie** [comment]
No worries, I am perfectly fine with double checking, I was sharing context

**Comment by yadij** [comment]
IIRC the HAVE_SQUID macro was for the kerberos helpers and purge tool when they were first bundled. The wrapper is no longer used by them, so it can go now.

File: include/version.h [hunk 1]
```diff
@@ -22,3 +25,4 @@
 #define APP_FULLNAME  PACKAGE "/" VERSION
 #endif
 
+#endif /* SQUID_VERSION_H */
```
Comment (yadij) [inline include/version.h:28]: git complains, missing newline before EOF
```suggestion
#endif /* SQUID_VERSION_H */

```

**Comment by kinkie** [comment]
@yadij mind unblocking this PR after the change? Thanks!

**Pull request closed without merging**